source/builtin_macros/src/syntax.rs

```diff
@@ -897,18 +926,23 @@ impl VisitMut for Visitor {
                         BinOp::Gt(..) => {
                             *expr = Expr::Verbatim(quote_spanned!(span => (#left).spec_gt(#right)));
                         }
-                        BinOp::Add(..) => {
+                        BinOp::Add(..) if !self.inside_bitvector => {
                             *expr =
                                 Expr::Verbatim(quote_spanned!(span => (#left).spec_add(#right)));
                         }
-                        BinOp::Sub(..) => {
+                        BinOp::Sub(..) if !self.inside_bitvector => {
                             *expr =
                                 Expr::Verbatim(quote_spanned!(span => (#left).spec_sub(#right)));
                         }
-                        BinOp::Mul(..) => {
+                        BinOp::Mul(..) if !self.inside_bitvector => {
                             *expr =
                                 Expr::Verbatim(quote_spanned!(span => (#left).spec_mul(#right)));
                         }
+                        BinOp::Add(..) | BinOp::Sub(..) | BinOp::Mul(..) => {
+                            *expr = Expr::Verbatim(
+                                quote_spanned!(span => compile_error!("Inside bit-vector assertion, use `add` `sub` `mul` for fixed-bit operators, instead of `+` `-` `*`, ")),
```

Comment:
Here, `add`, `sub`, and `mul` could be confused with the `.add`, `.sub`, and `.mul` *methods* on integer types, so we might want to add some additional information: `(see the functions builtin::add(left, right), builtin::sub(left, right), and builtin::mul(left, right))`.

Reply:
Ah, I agree. Sound great, thanks!